halo2_proofs/src/plonk.rs

```diff
@@ -44,17 +44,15 @@ use std::io;
 pub fn vk_read<C: SerdeCurveAffine, R: io::Read, ConcreteCircuit: Circuit<C::Scalar>>(
     reader: &mut R,
     format: SerdeFormat,
+    k: u32,
```

Comment:
It's unfortunate that we need to pass `k` because the serialized key already has `k`. https://github.com/privacy-scaling-explorations/halo2/blob/9eedeb5d6eb9a119d34307697e670b0bb043a5a5/halo2_backend/src/plonk.rs#L115C9-L115C38

But anyway, this is for the legacy API so I think it's OK.  But it makes me wonder, should we deprecate the legacy API at some point?  It would make our lives easier.

Reply:
Yes, I believe that we would need to deprecate the legacy API in near future.
But, I also believe that it should come after discussion with community.
In addition, I think we don't need more work with legacy API, later.